doc/developer-guides/hld/sriov-hld.rst

```diff
@@ -0,0 +1,303 @@
+   SR-IOV VF physical device instead *Vendor ID* since there is no valid
+   *Vendor ID* for SR-IOV VF physical device, the VF BARs are
+   initialized by its associated PF’s SR-IOV capabilities not PCI
+   standard BAR registers, MSIx mapping base address is also from PF’s
+   SR-IOV capabilities not PCI standard BAR registers.
+
+SR-IOV Passthrough VF Architecture In ACRN
+------------------------------------------
+
+.. figure:: images/sriov-image4.png
+   :align: center
+   :width: 900px
+   :name: SR-IOV-vf-passthrough
+
+   SR-IOV VF Passthrough Architecture In ACRN
+
+1. The SR-IOV VF device needs to bind PCI-stud driver instead
+   vendor-specific VF driver before passthrough the device.
+
+2. User configures acrn-dm boot parameter with passthrough SR-IOV VF
+   device. When UOS starts, acrn-dm invokes a hypercall to set the
+   *vdev-VF0* device in UOS.
+
+3. Hypervisor emulates *Device ID/Vendor ID* and *Memory Space Enable
+   (MSE*) in configuration space for an assigned SR-IOV VF device. The
+   assigned VF *Device ID* comes from its associated PF’s capability,
+   *Vendor ID* is same with PF’s *Vendor ID* and *MSE* is always set
+   when reading the SR-IOV VF device *CONTROL* register
+
+4. Vendor-specific VF driver in target VM probes the assigned SR-IOV VF
+   device.
+
+SR-IOV Initialization Flow
+--------------------------
+
+.. figure:: images/sriov-image5.png
+   :align: center
+   :width: 900px
+   :name: SR-IOV-init-flow
+
+   SR-IOV Initialization Flow
+
+When a SR-IOV capable device is initialized, all access to the
+configuration space will pass through to the physical device directly.
+SOS can identify all capabilities of the device from SR-IOV extended
+capability then create sysfs node for SR-IOV management.
+
+SR-IOV VF Enable Flow
+---------------------
+
+.. figure:: images/sriov-image6.png
+   :align: center
+   :width: 900px
+   :name: SR-IOV-enable-flow
+
+   SR-IOV VF Enable Flow
+
+Application enables n VF devices via a SR-IOV PF device *sysfs* node.
+Hypervisor intercepts all SR-IOV capability accesses and check
+*VF\_ENABLE* state, if the *VF\_ENABLE* is set, hypervisor creates n
+virtual devices after 100ms that VF physical devices have enough time to
+be created. SOS waits 100ms then only accesses the first VF device’s
+configuration space including *Class Code, Reversion ID, Subsystem
+Vendor ID, Subsystem ID,* SOS uses the first VF device information to
+initialize subsequent VF devices.
+
+SR-IOV VF Disable Flow
+----------------------
+
+.. figure:: images/sriov-image7.png
+   :align: center
+   :width: 900px
+   :name: SR-IOV-disable-flow
+
+   SR-IOV VF Disable Flow
+
+Application disables SR-IOV VF devices by writing zero to SR-IOV PF
+device *sysfs* node. Hypervisor intercepts all SR-IOV capability
+accesses and check *VF\_ENABLE* state, if *VF\_ENABLE* is clear,
+hypervisor makes VF virtual devices invisible from SOS that all accesses
```

Comment:
```suggestion
hypervisor makes VF virtual devices invisible from Service VM that all accesses
```

Reply:
What happens to the VF devices that have been passed-through to User VMs? (and is there any difference between RTVM and "normal" User VM)?